mysql: add MySQL InnoDB link to MySQL Compatibility doc (#21131)
pingcap/docs: mysql-compatibility.md

- [`CREATE RESOURCE GROUP`](/sql-statements/sql-statement-create-resource-group.md)
- [`DROP RESOURCE GROUP`](/sql-statements/sql-statement-drop-resource-group.md)
- [`ALTER RESOURCE GROUP`](/sql-statements/sql-statement-alter-resource-group.md)

> ## Differences from MySQL InnoDB

Author: ```suggestion
## Differences on pessimistic transaction (lock) with MySQL InnoDB
```

> For differences on pessimistic transaction (lock) with MySQL InnoDB, see [Differences from MySQL InnoDB](/pessimistic-transaction.md#differences-from-mysql-innodb).

Author: ```suggestion
For differences on pessimistic transaction (lock)  between TiDB and MySQL InnoDB, see [Differences from MySQL InnoDB](/pessimistic-transaction.md#differences-from-mysql-innodb).
```

> For differences on pessimistic transaction (lock)  between TiDB and MySQL InnoDB, see [Differences from MySQL InnoDB](/pessimistic-transaction.md#differences-from-mysql-innodb).

Author: ```suggestion
For differences on pessimistic transaction (lock) between TiDB and MySQL InnoDB, see [Differences from MySQL InnoDB](/pessimistic-transaction.md#differences-from-mysql-innodb).
```
